DSCResources/MSFT_xDSCWebService/SecureTLSProtocols.psm1

```diff
@@ -0,0 +1,48 @@
+# This module file contains a utility to test and set TLS protocol according best practices
+#
+# Copyright (c) Microsoft Corporation, 2016
+#
+
+# ============ Best Practice Security Settings Block =========
+$insecureProtocols            = @("SSL 2.0", "SSL 3.0", "TLS 1.0", "PCT 1.0", "Multi-Protocol Unified Hello")
+$secureProtocols              = @("TLS 1.1", "TLS 1.2")
+
+# ===========================================================
+
+function Test-Protocol
+{
+    foreach ($protocol in $insecureProtocols)
+    {
+        $registryPath = "HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\$protocol\Server"
+        if (($null -ne (Get-Item -Path $registryPath -ErrorAction SilentlyContinue)) -and ($null -ne (Get-ItemProperty -Path $registryPath)) -and ((Get-ItemProperty -Path $registryPath).Enabled -ne 0))
+        {
+            return $false
+        }
+    }
+    foreach ($protocol in $secureProtocols)
+    {
+        $registryPath = "HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\$protocol\Server"
+        if (($null -eq (Get-Item -Path $registryPath -ErrorAction SilentlyContinue -ErrorVariable ev)) -or ($null -eq (Get-ItemProperty -Path $registryPath)) -or ((Get-ItemProperty -Path $registryPath).Enabled -eq 0))
```

Comment:
Because when the path does not exist, get-item throws error. Yet, this is not an error case for the resource, it is only the testResult = $false case. Ignore all what I said about set. Only within logic of test, this is the case.

PS C:\Windows\system32> $ret = Get-Item -Path $registryPath -ErrorAction SilentlyContinue
PS C:\Windows\system32> $ret -eq $null
True
PS C:\Windows\system32> $ret = Get-Item -Path $registryPath
Get-Item : Cannot find path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\Server'
because it does not exist.
At line:1 char:8
+ $ret = Get-Item -Path $registryPath
+        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (HKLM:\SYSTEM\Cu...rotocols\Server:String) [Get-Item], ItemNotFoundExcep
   tion
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand